source/encoder.cpp

```diff
@@ -930,6 +939,9 @@ bool obsffmpeg::encoder::update(obs_data_t* settings)
 		_context->keyint_min = _context->gop_size;
 	}
 
+  if (!_hwinst)
+	  av_opt_set_int(_context->priv_data, "gpu", (int)obs_data_get_int(settings, ST_FFMPEG_GPU), 0);
```

Comment:
Set options on _context with AV_OPT_SEARCH_CHILDREN. It works better if the encoder and FFmpeg have the same property.

Reply:
This was also based upon the obs-ffmpeg standard plugin, but sure, I'll set the Search clildren option.

Reply:
Yeah it should point to _context. AV_OPT_SEARCH_CHILDREN makes it descend into _context->priv_data by itself, so when options move from encoder to ffmpeg or ffmpeg to encoder things still work as expected.